molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/AnnotationStep.java

```diff
@@ -0,0 +1,21 @@
+package org.molgenis.data.annotation.entity;
+
+import java.util.Collection;
+
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.Entity;
+
+/**
+ * A step to use in the Annotator. Each step has requirements on the annotated entity's metadata.
+ * 
+ * @author fkelpin
+ *
+ */
+public interface AnnotationStep
```

Comment:
Requires new name to better specify its function: Classes that implement can be asked; Which attributes do you need?